janusgraph-core/src/main/java/org/janusgraph/graphdb/management/ConfigurationManagementGraph.java

```diff
@@ -0,0 +1,310 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.graphdb.management;
+
+import org.janusgraph.graphdb.database.StandardJanusGraph;
+import org.janusgraph.core.schema.JanusGraphManagement;
+import org.janusgraph.graphdb.database.management.ManagementSystem;
+import org.janusgraph.graphdb.management.utils.ConfigurationManagementGraphNotEnabledException;
+import org.janusgraph.graphdb.management.utils.ConfigurationManagementGraphAlreadyInstantiatedException;
+import static org.janusgraph.core.schema.SchemaAction.ENABLE_INDEX;
+import static org.janusgraph.core.schema.SchemaAction.REGISTER_INDEX;
+import static org.janusgraph.core.schema.SchemaStatus.INSTALLED;
+import static org.janusgraph.core.schema.SchemaStatus.REGISTERED;
+import static org.janusgraph.core.schema.SchemaStatus.ENABLED;
+import org.janusgraph.core.PropertyKey;
+import org.janusgraph.core.schema.JanusGraphIndex;
+import org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration;
+
+import org.apache.tinkerpop.gremlin.structure.Graph;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+import org.apache.tinkerpop.gremlin.structure.T;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal;
+
+import org.apache.commons.configuration.MapConfiguration;
+import org.apache.commons.configuration.ConversionException;
+import com.google.common.base.Preconditions;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.NoSuchElementException;
+import java.util.Map;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+public class ConfigurationManagementGraph {
+
+    private static final Logger log = LoggerFactory.getLogger(ConfigurationManagementGraph.class);
+    private static ConfigurationManagementGraph instance = null;
+    public static final String PROPERTY_GRAPH_NAME = GraphDatabaseConfiguration.GRAPH_NAME.toStringWithoutRoot();
+    public static final String PROPERTY_CREATED_USING_TEMPLATE = "Created_Using_Template";
+    private static final String VERTEX_LABEL = "Configuration";
+    private static final String GRAPH_NAME_INDEX = "Graph_Name_Index";
+    private static final String PROPERTY_TEMPLATE = "Template_Configuration";
+    private static final String TEMPLATE_INDEX = "Template_Index";
+
+    private final StandardJanusGraph graph;
+
+    /**
+     * This class allows you to create/update/remove configuration objects used to open a {@link JanusGraph}.
```

Comment:
Change `{@link JanusGraph}` to just "graph" or else import/use fully-qualified name.